medschool/src/parse.rs

```diff
@@ -0,0 +1,286 @@
+            None
+        })
+        .collect()
+}
+
+/// Digs into the generics of a field ([`syn::ItemStruct`] only), trying to get the last
+/// [`syn::PathArguments`], which (hopefully) contains the concrete type we care about.
+///
+/// Extracts `Foo` from `foo: Option<Vec<{Foo}>>`.
+///
+/// Doesn't handle generics of generics though, so if your field is `baz: Option<T>` we're going
+/// to be assigning this field type to be the string `"T"` (which is probably not what you
+/// wanted).
+#[tracing::instrument(level = "trace", ret)]
+pub fn get_ident_from_field_skipping_generics(type_path: TypePath) -> Option<Ident> {
+    // welp, this path probably contains generics
+    let segment = type_path.path.segments.last()?;
+
+    let mut current_argument = segment.arguments.clone();
+    let mut inner_type = None;
+
+    // keep digging into the `PathArguments`
+    while let syn::PathArguments::AngleBracketed(generics) = &current_argument {
+        // go directly to the last piece of generics, skipping lifetimes
+        match generics.args.last()? {
+            // finally have something that resembles a type, but might be an `Option`, so
+            // we have to go deeper!
+            syn::GenericArgument::Type(Type::Path(generic_path)) => {
+                // that's it, we've reached the final type
+                inner_type = match generic_path.path.segments.last() {
+                    Some(t) => Some(t.ident.clone()),
+                    None => break,
+                };
+
+                current_argument = generic_path.path.segments.last()?.arguments.clone();
+            }
+            _ => break,
+        }
+    }
+
+    inner_type
+}
+
+/// Converts a [`syn::ItemMod`] into a [`PartialType`].
+fn parse_item_mod(item_mod: syn::ItemMod) -> Option<PartialType> {
+    let thing_docs_untreated = docs_from_attributes(item_mod.attrs);
+    Some(PartialType {
+        ident: item_mod.ident.to_string(),
+        docs: thing_docs_untreated,
+        fields: Default::default(),
+    })
+}
+
+/// Converts a [`syn::ItemEnum`] into a [`PartialType`].
+fn parse_item_enum(item: syn::ItemEnum) -> Option<PartialType> {
+    let thing_docs_untreated = docs_from_attributes(item.attrs);
+    let is_internal = thing_docs_untreated
+        .iter()
+        .any(|doc| doc.contains(r"<!--${internal}-->"));
+
+    // We only care about types that have docs.
+    (!thing_docs_untreated.is_empty() && !is_internal).then(|| PartialType {
+        ident: item.ident.to_string(),
+        docs: thing_docs_untreated,
+        fields: Default::default(),
+    })
+}
+
+/// Converts a [`syn::ItemStruct`] into a [`PartialType`].
+fn parse_item_struct(item: syn::ItemStruct) -> Option<PartialType> {
+    let mut docs = docs_from_attributes(item.attrs);
+
+    if docs.is_empty() {
+        return None;
+    }
+
+    for doc in docs.iter_mut() {
+        // removes docs that we don't want in `configuration.md`
+        if doc.contains(r"<!--${internal}-->") {
```

Comment:
Instead of checking for this on every call site, if you turn `docs_from_attributes` to return an `Option<Vec>`, you could check for this there, and return `None`, everywhere else would become just `let mut docs = docs_from_attributes(whatever)?;`.